### PR TITLE
handle model property conflicts when format is "date" or "datetime"

### LINKED ISSUE #1220: Model file cannot compile when a string field has both date format and maxLength
## Problem statement

When the swagger file has a string field with both date format and maxLength, the generated model file does not compile.

In this example (adapted from [petstore](https://github.com/OAI/OpenAPI-Specification/blob/master/examples/v2.0/json/petstore-simple.json), the field "birth" is string, has date format, and a max length of 10.

We should call m.Birth.String() rather than string(m.Birth) in new_pet.go.

## Swagger specification

```json
{
  "swagger": "2.0",
  "info": {
    "version": "1.0.0",
    "title": "Swagger Petstore"
  },
  "host": "petstore.swagger.io",
  "basePath": "/api",
  "schemes": [
    "http"
  ],
  "consumes": [
    "application/json"
  ],
  "produces": [
    "application/json"
  ],
  "paths": {
    "/pets": {
      "post": {
        "operationId": "addPet",
        "produces": [
          "application/json"
        ],
        "parameters": [
          {
            "name": "pet",
            "in": "body",
            "description": "Pet to add to the store",
            "required": true,
            "schema": {
              "$ref": "#/definitions/NewPet"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "pet response",
            "schema": {
            }
          },
          "default": {
            "description": "unexpected error",
            "schema": {
            }
          }
        }
      }
    }
  },
  "definitions": {
    "NewPet": {
      "type": "object",
      "required": [
        "name"
      ],
      "properties": {
        "name": {
          "type": "string"
        },
        "birth": {
          "type": "string",
          "format": "date",
          "maxLength": 10
        }
      }
    }
  }
}
```

## Steps to reproduce

```
$ swagger generate client --spec apis/test.json
$ go build client/swagger_petstore_client.go 
# github.com/neighborly/experiments/pershing-demo/models
models/new_pet.go:60:54: cannot convert m.Birth (type strfmt.Date) to type string
```

## Environment
swagger version: dev
go version: 1.9.1
OS: OSX

### PR DESCRIPTION
**Pull request by kreamyx**
fixes #1220

### CODE REVIEW

File: generator/model.go [hunk 1]
```diff
@@ -543,8 +543,23 @@ func hasValidations(model *spec.Schema, isRequired bool) (needsValidation bool,
...
+	case "datetime":
+		model.MinLength = nil
+		model.MaxLength = nil
+		// more cases should be inserted here if they arise
+	}
+}
+
 func (sg *schemaGenContext) schemaValidations() sharedValidations {
 	model := sg.Schema
+	// resolve any conflicting properties if the model has a Fromat
```
Comment (gregmarr) [inline generator/model.go]: typo: Fromat

File: generator/model.go [hunk 2]
```diff
@@ -543,8 +543,23 @@ func hasValidations(model *spec.Schema, isRequired bool) (needsValidation bool,
 	return
 }
 
+// handleFormatConflicts handles all conflicting model properties when a format is set
+func handleFormatConflicts(model *spec.Schema) {
+	switch model.Format {
+	case "date":
+		model.MinLength = nil
+		model.MaxLength = nil
+	case "datetime":
```
Comment (gregmarr) [inline generator/model.go]: Since "date" and "datetime" have the same conflicts, you can combine the cases:

    case "date", "datetime":

Comment (gregmarr) [reply generator/model.go]: Probably "uuid" as well.

File: generator/model.go [hunk 3]
```diff
@@ -543,8 +543,20 @@ func hasValidations(model *spec.Schema, isRequired bool) (needsValidation bool,
 	return
 }
 
+// handleFormatConflicts handles all conflicting model properties when a format is set
+func handleFormatConflicts(model *spec.Schema) {
+	switch model.Format {
+	case "date", "datetime", "uuid":
```
Comment (casualjim) [inline generator/model.go]: also: bsonobjectid, base64 and duration

none of those are actually strings so the validations don't apply to them

**Pull request merged**